mtags/src/main/scala-3/scala/meta/internal/pc/CompletionProvider.scala

```diff
@@ -0,0 +1,169 @@
+package scala.meta.internal.pc
+
+import scala.collection.mutable
+import scala.meta.internal.pc.IdentifierComparator
+
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.core.Flags._
+import dotty.tools.dotc.core.Names._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.core.StdNames._
+import dotty.tools.dotc.core.Types._
+import dotty.tools.dotc.interactive.Completion
+import dotty.tools.dotc.transform.SymUtils._
+import dotty.tools.dotc.util.Spans
+import dotty.tools.dotc.util.SrcPos
+import dotty.tools.dotc.util.{NameTransformer, NoSourcePosition, SourcePosition}
+
+class CompletionProvider(
+    pos: SourcePosition,
+    ctx: Context
+) {
+    implicit val context: Context = ctx
+      
+    def completions() : List[Completion]
+        = Completion.completions(pos)._2.filterInteresting.sorted(completionOrdering(ctx, pos))
+
+    implicit class CompletionExtension(val c: Completion) {
+      // completionItem method either way takes only head symbol, so why can safely ignore the rest
+      val sym: Symbol = c.symbols.head
+    }
+
+    implicit class CompletionListExtension(val l: List[Completion]) {
+      val filterInteresting: List[Completion] = {
+        val isSeen = mutable.Set.empty[String]
+        val buf = List.newBuilder[Completion]
+        implicit class SrcPosExtension(val s: SrcPos) {
+          def isAfter(s1: SrcPos) = s.sourcePos.point > s1.sourcePos.point
```

Comment:
It's actually used in one place, maybe we can inline it?

Reply:
Actually, it's now used in one more place